tests/bluechi_test/machine.py

```diff
@@ -312,18 +312,33 @@ def __init__(
         self.copy_machine_lib()
 
 
-class BluechiControllerMachine(BluechiMachine):
-    def __init__(
-        self, name: str, client: Client, ctrl_config: BluechiControllerConfig
-    ) -> None:
-        super().__init__(name, client)
+class BluechiControllerMachine(BluechiAgentMachine):
+
+    agent_uds_address = "unix:path=/run/bluechi/bluechi.sock"
+    agent_uds_name = "node-local"
+
+    def __init__(self, name: str,
+                 client: Client,
+                 ctrl_config: BluechiControllerConfig,
+                 agent_config: BluechiAgentConfig = BluechiAgentConfig("agent.conf"),
```

Comment:
This should default to `None` since there might be tests where we don't want a local bluechi agent. 

I'd create a `self.bluechi_agent_on_ctrl_machine_cfg` (or some better name) on the `BluechiTest` class, also initializing it with `None`. This way we can use the `bluechi_node_default_config` fixture in the respective tests, modify it if needed and pass it down. This requires updates to most (if not all) tests, of course.

Reply:
> This should default to None since there might be tests where we don't want a local bluechi agent.

Can this be achieved in this way, just set `no` per test(those don't need a local agent). e.g:
```
os.environ["LOCAL_AGENT"] = "no" # default value is "yes"
def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
```

so most of the test cases do not need to be modified.

Reply:
Using env variables for test specific configuration might affect subsequent tests if not cleaned up, so I'd rather avoid it.

Reply:
I'd like to avoid any per test configuration regarding agent on controller node